### DISCUSSION TITLE
Could not find Git; can not register Git LFS.

### DISCUSSION THREAD

**Discussion by Invidelis** [git-lfs/git-lfs#5031] [Q&A]
hello, the title says it all for me.

I downloaded the GIT LFS v3.2.0 and wanted to install it. I opened it as administrartor and got the following error:

`Could not find Git; can not register Git LFS.`

![image](https://user-images.githubusercontent.com/11723315/171229863-b4b49563-0560-4948-aee3-5ae55b11e340.png)

Anyone had this error before?  And yes I have GitHub desktop installed.

**Comment by Nathan-Amiri** [comment]
I had the same issue--you need to install Git in addition to Github desktop!
Here's the link: https://gitforwindows.org/

**Reply by pickyusernamepicker** [reply]
Nope, something's the matter with Git LFS 3.2.0. I've got Git for Windows 2.36.1 and Git LFS 3.1.4, never had any problems. Now I get the same message as OP when trying to upgrage Git LFS to 3.2.0. I tried reinstalling Git for Windows, but it didn't help.

**Comment by chrisd8088** [comment]
Git needs to be available in `PATH`; otherwise, as of Git 3.2.0, you'll see the warning above as result of the changes in #4925.

If you can provide a reproducible case or details on where your Git binary is installed and what is in your `PATH` environment variables (system/machine and user), etc., that might help further troubleshoot the issue.  For instance, can you run Git from the command line just as `git` or `git.exe` without providing a full path?

**Reply by pickyusernamepicker** [reply]
Ah, sorry. I somehow thought that Git for Windows has two separate options, one for adding itself to PATH, the other for its availability from the command line. But this is not the case:
![components 2 36 1](https://user-images.githubusercontent.com/34836588/173807051-24996c39-2aab-423f-acc0-efe16e55664c.PNG)
![cli 2 36 1](https://user-images.githubusercontent.com/34836588/173807079-85cf650c-e787-45df-8f43-b518470c48a6.PNG)
I reinstalled Git without "Use Git from Git Bash only" and the problem is gone. Thank you

**Reply by dscho** [reply]
I guess we could do the same in Git LFS' installer as we do in Git for Windows' installer: [look at `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Git_is1`](https://github.com/git-for-windows/build-extra/blob/4284c9b083afade545cd4115fb654ce5c6ddc582/installer/install.iss#L2637-L2666):

![image](https://user-images.githubusercontent.com/127790/173852660-9dc7bfb1-0073-41ba-8ef3-9dd6683c0cc0.png)

If we do not find `git.exe` in the `PATH`, we would see whether `Inno Setup CodeFile: Path Option` is [`BashOnly`](https://github.com/git-for-windows/build-extra/blob/4284c9b083afade545cd4115fb654ce5c6ddc582/installer/install.iss#L2086) and if it is, look at `Inno Setup: App Path` and see whether we can use `<App Path>\cmd\git.exe` instead.

@chrisd8088 what do you think?

**Reply by chrisd8088** [reply]
That seems reasonable; it would be a nice addition for the next release.  Thanks!

**Reply by dscho** [reply]
Here you are: https://github.com/git-lfs/git-lfs/pull/5048

**Comment by MegacosmGames** [comment]
I have the same issue as [(https://github.com/git-lfs/git-lfs/discussions/5031#discussion-4110835)]. I am new to git so, can someone dumb down the solution for me? What I understood is that I need to download GIT for windows and have the above selections when installing?

**Reply by chrisd8088** [reply]
The official patch for this problem, from #5048, will be in the next Git LFS release, slated for a couple of weeks from now.

In the meantime, if you reinstall Git for Windows and do **not** choose the **"Use Git from Git Bash only"** option, that should help.

**Comment by chrisd8088** [comment]
For anyone tracking this discussion, please try this pre-release version of Git for Windows and see if it fixes the problem when choosing the **"Use Git from Git Bash only"** option:

https://github.com/git-for-windows/git/releases/tag/v2.39.0-rc2.windows.1

Thank you, and please report if it works or if the problem persists!

**Reply by Stygmire** [reply]
Having the same issue as OP but with with Windows 3.3.0.

Because I am following tutorials, and they seem to imply only using GithubDesktop is required, can you (or someone else here) confirm if the desktop-version of Git is enough, or I am required to install the Git for Windows as well as GithubDesktop?

I did check my path variable and the GithubDesktop is listed, and 'gh' commands work at a regular command-prompt.

'git' commands do not work...

Plz help.  :(

**Reply by bk2204** [reply]
I don't know if GitHub Desktop comes with Git.  _If_ it does, it will also come with Git LFS, so you should not need to install Git LFS manually in such a case.

Typically, if you just install Git for Windows, then Git LFS comes with it, so I'd recommend that as the approach if possible and avoid installing Git LFS separately.

**Comment by Stygmire** [comment]
TY sir, your input is appreciated!! :D

**Comment by TechQuery** [comment]
WinGet users may find some helps here: https://github.com/git-lfs/git-lfs/issues/5147#issuecomment-2542269120

**Comment by AugustYadon** [comment]
Just thought I'd chime in and say I use Github Desktop for my projects normally, and I am also getting this same issue in 2025 when trying to install LFS via https://docs.github.com/en/repositories/working-with-files/managing-large-files/installing-git-large-file-storage

I reinstalled Git as suggested by @Nathan-Amiri  "Here's the link: https://gitforwindows.org/" and that seems to have fixed it. 
Also selected "Use Git from Git Bash Only".

I was just commenting because it was unclear with some of the posts suggesting that this would be automatically fixed in a 2022 release of github desktop. I believe LFS was also installed this time along with Git, so that is nice. Anyway just thought I'd report back what worked for me recently.

Edit: Just wanted to say, even though it installed LFS, I still had to open command line and type "git install lfs" when it was done to initialize LFS one time!
